### ISSUE TITLE
Grabbing people is really hard!

### ISSUE THREAD

**Issue by FenodyreeAv** [PolarisSS13/Polaris#6427]
#### Brief description of the issue
TL:DR 
-  You've got over a 99% chance to break out a bluegrab before it's upgraded to redgrab!
-  Upgrading a blue pin, to a red pin increases the resist chance from 15% to 45%.

Over time I'd noticed that it's been getting harder and harder to handcuff someone.
Nothing changed with handcuffs, so the reason for that is grabcode.

First off, you can just walk out of a yellowgrab. This is pretty big. Even confused, and super slow you can walk away before the grab gets to blue. Also, resist is on the SAME COOLDOWN as the grab. So if you spam resist, it is impossible to be bluegrabbed.

Specifically, the ease of resisting out of an aggressive (blue) grab.
The base chance, for two humans to break out is 60%.
This is changed by the INCAPACITATION_KNOCKDOWN flag, to 15% to break out.

This flag doesn't check hallos, confusion, blindness or injuries, only whether the person is on the floor or not. All of security's weapons inflict the above conditions (honorable exception being pepperspray, which only lasts 5 ticks), meaning an awake person will have a 60% chance to break out of your grab.

You can resist every 8 ticks, it takes 40 ticks to upgrade a grab, meaning you've got five 60% chances to break it.

Then it takes at least 30 ticks to put on handcuffs. Giving you at least another four 45% chances to break the grab. (Because if you've moved at in the last 3 seconds it makes the grab easier to break, and because it only checks if you're STUNNED. A condition which sec can't inflict).

1 - (1 - 0.6)^6 = 0.995904
1 - (1 - 0.45)^4 = 0.90849375
Or in other words, you've got a 99.5% chance of breaking the bluegrab.
And 90.8% chance to breaking out of the redgrab.

But, lets examine the best case scenario. You're quick, and they're slow. You manage to get the bluegrab while they're still peppersprayed AND switch your pepperspray out for a pair of handcuffs, and click them again, so they are pinned to the ground.

 Then they have 30 ticks to be able to resist at 15%.
1 - (1 - 0.15)4 = 0.47799375
That's 47.7% chance of failure for a skilled player under **ideal** conditions.

#### What you expected to happen
A dude who's blind, in massive pain, stumbling around confused, with both knees broken, to be pretty easy to grab.

#### What actually happened
You always have to put someone into paincrit to grab, or handcuff them.

#### Steps to reproduce
Attempt to handcuff a man who's spamming resist, without being able to paincrit them.

#### Additional info:
 - **Server Revision**: Found using the "Show Server Revision" verb under the OOC tab.
 - **Anything else you may wish to add** (Location if it's a mapping issue, etc)

**Comment by Wickedtemp** [comment]
I feel like making a slightly longer cooldown on Resist might fix things a bit, or make each subsequent resist less effective, so spamming isn't as good.

**Comment by Atermonera** [comment]
Or perhaps some combination of the two? Make it so breakout chance is affected by time since last resist (up to a maximum), such that mashing the button is pretty much a waste of time, but the chances of escape don't become so miniscule that once you get past a certain number of attempts you may as well start mashing it because your only hope is to annoy your captor so much that they let you go themselves.
